### PR TITLE
Support Fixed-Sized Lists

### PR DESCRIPTION
**Pull request by Benjamin-Philip**
This commit adds support for the Fixed-Sized List Layout.

### CODE REVIEW

**Comment by Benjamin-Philip** [comment]
I've made some changes to the type system so that a shorthand (atoms) and longhand (tuples) syntax for declaring types exist for better compatibility with Nx.

File: src/serde_arrow_type.erl [hunk 1]
```diff
@@ -1,19 +1,142 @@
...
+%%      <td>Float</td>
+%%      <td>`f'</td>
+%%      <td>`f32'</td>
+%%  </tr>
+%%  <tr>
+%%      <td>Binaries</td>
+%%      <td>N/A</td>
+%%      <td>`bin'</td>
+%%  </tr>
+%% </table>
```
Comment (Benjamin-Philip) [inline src/serde_arrow_type.erl:76]: This table results in invalid output with ExDoc, but has have valid output with EDoc. I will create an new in `rebar3_ex_doc`. How much I would give to write docs in AsciiDoc or even Markdown!

**Review by josevalim — APPROVED** [review]

**Comment by Benjamin-Philip** [comment]
@josevalim and @tsloughter, I've added initial support for Fixed-Size Lists. Couple of things left to do:

1. Test if we crash on invalid input
2. Write docs

**Review by josevalim — APPROVED** [review]

File: src/serde_arrow_type.erl (hunk 1 above)
Comment (Benjamin-Philip) [reply src/serde_arrow_type.erl:76]: I've created an issue for this: https://github.com/starbelly/rebar3_ex_doc/issues/47

**Pull request merged**